src/python/WMCore/WorkQueue/DataLocationMapper.py

```diff
@@ -246,6 +246,8 @@ def updatePileupLocation(self, fullResync = False):
             for data, locations in dataMapping.items():
                 elements = self.backend.getElementsForPileupData(data)
                 for element in elements:
+                    if element.get('NoLocationUpdate', False):
+                        continue
```

Comment:
I think I know why, because the PU data location makes no difference at this level since it's handled differently (job creation does not depend on PU data, and it's created by PileupFetcher.py querying directly dbs).

The only thing we need to make sure is that elements will get replicated (and for this case, don't waste time updating pileup location).